front-end/src/app/shared/interfaces/transaction-type.interface.ts

```diff
@@ -11,9 +13,18 @@ export interface TransactionType {
   isDependentChild: boolean; // When set to true, the parent transaction is used to generate UI form entry page
   title: string;
   schema: JsonSchema; // FEC validation JSON schema
-  transaction: Transaction | undefined;
-  parentTransaction: Transaction | undefined;
-  childTransactionType: TransactionType | undefined;
+  transaction?: Transaction;
+  parentTransaction?: Transaction;
+  childTransactionType?: TransactionType;
+  subTransactionTypes?: ScheduleATransactionTypes[];
+  navigationControls?: TransactionNavigationControls;
   contributionPurposeDescripReadonly(): string; // Dynamically generates the text in the CPD field
   getNewTransaction(): Transaction; // Factory method to create a new Transaction object with default property values for this transaction type
 }
+
+export function isNewTransaction(transaction?: Transaction): boolean {
+  return !transaction?.id;
+}
+export function hasContact(transaction?: Transaction): boolean {
+  return !transaction?.contact;
+}
```

Comment:
Rather than having these functions defined as separate in the interface file, would the code be better organized if we crated a BaseTransactionModel that contained these methods and SchATransaction model could inherit the methods? BaseTransactionModel would extend BaseModel.

Reply:
yeah wow that makes a lot more sense